src/history/historyTracker.ts

```diff
@@ -441,6 +442,24 @@ export class HistoryTracker {
       editor: isUppercaseMark ? vscode.window.activeTextEditor : undefined,
     };
     this.putMarkInList(newMark);
+    this.updateMarksDecorators();
+  }
+
+  /**
+   * Update marks decorations in the gutter
+   */
+  private updateMarksDecorators(): void {
+    if (!this.vimState.editor) {
+      return;
+    }
+
+    let decorators: vscode.Range[] = [];
+    for (const mark of this.getMarks()) {
```

Comment:
nit: can you change this to use `map` rather than a for loop?

Reply:
done